### PR TITLE
Syslog support

### PR DESCRIPTION
**Pull request by vvviperrr**
syslog logging is useful in embedded applications, e.g in buildroot software

### CODE REVIEW

**Comment by rfjakob** [comment]
Hi, thanks for the PR!

But first I have to ask a stupid question: Why is this better than piping stdout and stderr to logger(1)?

**Comment by vvviperrr** [comment]
> Hi, thanks for the PR!
> 
> But first I have to ask a stupid question: Why is this better than piping stdout and stderr to logger(1)?

Hi. The main reason - support different syslog priorities. Sure, we can achieve this via piping earlyoom's stdout & stderr in different logger instances with different facilities\levels, but this trick is really ugly.
something like this (example from busybox sysctl script)
```
	xargs -0 -r -n 1 readlink -f | {
		prog_status="OK"
		while :; do
			read -r file || {
				echo "$prog_status" >&4
				break
			}
			echo "* Applying $file ..."
			/sbin/sysctl -p "$file" $SYSCTL_ARGS || prog_status="FAIL"
		done 2>&1 >&3 | /usr/bin/logger -t sysctl -p kern.err
	} 3>&1 | /usr/bin/logger -t sysctl -p kern.info
```

the things become more complicated, when u need start earlyoom via start-stop-daemon.

so patching earlyoom code is just handy (by adding syslog param in default conf file) & portable solution.

File: msg.c [hunk 1]
```diff
@@ -60,27 +119,43 @@ static void color_log(FILE* f, const char* color, const char* fmt, va_list vl)
...
+    if (use_syslog) {
+        return vsyslog(log_level_syslog(level), fmt, vl);
+    }
+
+    color_log(log_level_stream(level), log_level_color(level), fmt, vl);
+}
+
+void earlyoom_syslog_init(void)
+{
+    if (!use_syslog) {
```
Comment (rfjakob) [inline msg.c:133]: Do we need this `if`?

File: main.c [hunk 2]
```diff
@@ -123,7 +123,7 @@ int main(int argc, char* argv[])
     meminfo_t m = parse_meminfo();
 
     int c;
-    const char* short_opt = "m:s:M:S:kingN:dvr:ph";
+    const char* short_opt = "m:s:M:S:kingN:Ldvr:ph";
```
Comment (rfjakob) [inline main.c:126]: Please use `--syslog` instead of `-L` to be more explicit

**Review by rfjakob — CHANGES_REQUESTED** [review]
Code looks clean but a little over-engineered :)

1. I don't think `enum LOG_LEVEL` is needed. Just use the syslog levels directly.
2. I don't think `log_level_stream`, `log_level_color`, `log_level_syslog` are needed. Instead, add `stream`, `color` parameters to logger.

File: msg.c (hunk 1 above)
Comment (vvviperrr) [reply msg.c:77]: just double-init check, which is does not leads to any problems, but anyway.

**Comment by rfjakob** [comment]
I merged this, thanks. Two more things if you have time:

1) Can you add `--syslog` to https://github.com/rfjakob/earlyoom/blob/master/MANPAGE.md

2) The startup messages still go to stdout, seems to be they should be going to syslog as well:

```
$ ./earlyoom --syslog
earlyoom v1.7-36-g848653f
mem total: 23881 MiB, user mem total: 21593 MiB, swap total: 8191 MiB
sending SIGTERM when mem <= 10.00% and swap <= 10.00%,
        SIGKILL when mem <=  5.00% and swap <=  5.00%
```

**Pull request merged**